### PR TITLE
hackertyper: init at 1.1

### PR DESCRIPTION
**Pull request by marius851000**
###### Motivation for this change
Wanted to test this and train at package creation

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/misc/hackertyper/default.nix
Comment (etu) [inline pkgs/tools/misc/hackertyper/default.nix]: So the commit id hash from version should be copied here.

File: pkgs/tools/misc/hackertyper/default.nix
Comment (etu) [inline pkgs/tools/misc/hackertyper/default.nix]: Missing homepage, probably link to github unless they have another website linked somewhere

**Review by etu — CHANGES_REQUESTED** [review]
Some small changes suggested, good start for the first package :+1:

File: maintainers/maintainer-list.nix
Comment (marius851000) [reply maintainers/maintainer-list.nix]: I don't know how I can do this.

Comment (etu) [reply maintainers/maintainer-list.nix]: I think that what you want to do is to squash your two commits into one commit. Then amend that commit to not include your addition to `maintainer-list.nix`. And then you can make a new commit with the change to the `maintainer-list.nix`.

And if you then wish you can re-order the commits so the `maintainer-list.nix` comes before the creation of the derivation file.

**Comment by marius851000** [comment]
Ok, I finished to correct the error. Thank etu

File: pkgs/tools/misc/hackertyper/default.nix
Comment (etu) [inline pkgs/tools/misc/hackertyper/default.nix]: Usually we go with YYYYMMDD since a format like MMDDYYYY may not always be an incrementing in number size, not that it matters that much for nix, it's more a convention.

File: pkgs/tools/misc/hackertyper/default.nix
Comment (etu) [inline pkgs/tools/misc/hackertyper/default.nix]: This line needs to be
```
meta = with stdenv.lib; {
```

Otherwise the licenses and maintainers aren't available below. So this breaks eval of nixpkgs.

**Comment by dtzWill** [comment]
Hash is wrong, probably wasn't updated when moving from 1.1 to the latest git.  Or however it happened, it seems wrong now--`nix-prefetch-url -A hackertyper.src` from your nixpkgs source tree will produce the correct hash.

**Comment by Lassulus** [comment]
Could re rebase this on the current master? merging it now would result in a dual entry in the maintainer-list.nix

File: pkgs/tools/misc/hackertyper/default.nix
Comment (Lassulus) [inline pkgs/tools/misc/hackertyper/default.nix]: We probably will remove the unquoted url syntax in the near future: NixOS/rfcs#45
And sorry for the delay, I didn't get a notification when you force pushed. Can you ping me with @Lassulus after the next push and I will merge this :)

**Review by etu — APPROVED** [review]

File: pkgs/tools/misc/hackertyper/default.nix
Comment (marius851000) [reply pkgs/tools/misc/hackertyper/default.nix]: Done, @Lassulus . I think this RFC is a good idea.

**Pull request merged**
